### PR TITLE
Add Release Notes for Active Storage

### PR DESCRIPTION
**Pull request by alexcameron89**
### Summary
I've added the relevant release notes from the CHANGELOG and deprecations in Active Storage for Rails 6.

Cc @prathamesh-sonpatki and @georgeclaghorn

### CODE REVIEW

File: guides/source/6_0_release_notes.md [hunk 1]
```diff
@@ -605,10 +605,68 @@ Please refer to the [Changelog][active-storage] for detailed changes.
...
 ### Deprecations
 
+*   Deprecate  `config.active_storage.queue` in favor of `config.active_storage.queues.analysis`
+    and `config.active_storage.queues.purge`.
+    ([Pull Request](https://github.com/rails/rails/pull/34838))
+
+*   Deprecate `ActiveStorage::Downloading` in favor of `ActiveStorage::Blob#open`.
+    ([Commit](https://github.com/rails/rails/commit/ee21b7c2eb64def8f00887a9fafbd77b85f464f1))
+
+*   Deprecate using `mini_magick` for generating image variants in favor of
```
Comment (georgeclaghorn) [inline guides/source/6_0_release_notes.md]: Tiny nit: I’d say “Deprecate using MiniMagick *directly*” or similar. You can still use MiniMagick, you just have to do it through ImageProcessing.

**Review by georgeclaghorn — COMMENTED** [review]
:heart:

**Comment by alexcameron89** [comment]
@georgeclaghorn Thank you for your review! I updated the commit with the clarification.

**Review by prathamesh-sonpatki — APPROVED** [review]

File: guides/source/6_0_release_notes.md [hunk 2]
```diff
@@ -605,10 +605,68 @@ Please refer to the [Changelog][active-storage] for detailed changes.
...
+*   Add the ability to reflect on defined attachments using the existing
+    ActiveRecord reflection mechanism.
+    ([Pull Request](https://github.com/rails/rails/pull/33018))
+
+*   Add `ActiveStorage::Blob#open`, which downloads a blob to a tempfile on disk
+    and yields the tempfile.
+    ([Commit](https://github.com/rails/rails/commit/ee21b7c2eb64def8f00887a9fafbd77b85f464f1))
+
+*   Support streaming downloads from Google Cloud Storage. Require version 1.11+
+    of the google-cloud-storage gem.
```
Comment (abhaynikam) [inline guides/source/6_0_release_notes.md]: I think we should quote the gem name here like it is done on this line -> https://github.com/rails/rails/pull/36149/files#diff-c92b5b23d4a27f4187e0532a81ff301fR664

```suggestion
    of the `google-cloud-storage` gem.
```

File: guides/source/6_0_release_notes.md [hunk 3]
```diff
@@ -605,10 +605,68 @@ Please refer to the [Changelog][active-storage] for detailed changes.
...
 
-*   Updating an attached model via `update` or `update!` with, say,
-    `@user.update!(images: [ … ])` now replaces the existing images instead of merely adding to them.
+*   Add support for generating BMP image variants.
+    ([Pull Request](https://github.com/rails/rails/pull/36051))
+
+*   Add support for generating TIFF image variants.
+    ([Pull Request](https://github.com/rails/rails/pull/34824))
+
+*   Add support for generating Progress JPEG image variants.
```
Comment (abhaynikam) [inline guides/source/6_0_release_notes.md]: According to the changelog it should be `Progress` -> `progressive`. 

```suggestion
*   Add support for generating progressive JPEG image variants.
```

File: guides/source/6_0_release_notes.md [hunk 4]
```diff
@@ -605,10 +605,68 @@ Please refer to the [Changelog][active-storage] for detailed changes.
...
+
+*   Add a generic `ActiveStorage::Error` class that Active Storage exceptions inherit from.
+    ([Commit](https://github.com/rails/rails/commit/18425b837149bc0d50f8d5349e1091a623762d6b))
+
+*   Persist uploaded files assigned to a record to storage when the record
+    is saved instead of immediately.
+    ([Pull Request](https://github.com/rails/rails/pull/33303))
+
+*   Add the ability to reflect on defined attachments using the existing
+    ActiveRecord reflection mechanism.
```
Comment (abhaynikam) [inline guides/source/6_0_release_notes.md]: Should this be `Active Record` as per https://edgeguides.rubyonrails.org/api_documentation_guidelines.html#wording ?

```
The proper names of Rails components have a space in between the words, like "Active Support". ActiveRecord is a Ruby module, whereas Active Record is an ORM.
```


```suggestion
    Active Record reflection mechanism.
```

**Pull request merged**